lib/rust/ensogl/examples/built-in-shapes/src/lib.rs

```diff
@@ -0,0 +1,329 @@
+//! Example scene showing the usage of built-in high-level shapes.
+
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+#![allow(clippy::bool_to_int_with_if)]
+#![allow(clippy::let_and_return)]
+
+use ensogl_core::display::shape::*;
+use ensogl_core::display::world::*;
+use ensogl_core::prelude::*;
+
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_core::display::navigation::navigator::Navigator;
+use ensogl_core::display::object::ObjectOps;
+
+
+
+// ==============
+// === Shapes ===
+// ==============
+
+mod rectangle {
+    use super::*;
+    ensogl_core::shape! {(
+        style: Style,
+        color: Vector4,
+        corner_radius: f32,
+        inset: f32,
+        border: f32,
+        border_color: Vector4,
+        clip: Vector2,
+    ) {
+            // === Canvas ===
+            let canvas_width = Var::<Pixels>::from("input_size.x");
+            let canvas_height = Var::<Pixels>::from("input_size.y");
+
+            // === Clip ===
+            let canvas_clip_height_diff = &canvas_height * (clip.y() * 2.0);
+            let canvas_clip_width_diff = &canvas_width * (clip.x() * 2.0);
+            let canvas_height = canvas_height + &canvas_clip_height_diff;
+            let canvas_width = canvas_width + &canvas_clip_width_diff;
+
+            // === Body ===
+            let inset2 = (&inset * 2.0).px();
```

Comment:
I'll leave inset here, as we discussed, as this is term used in CSS and Swift. Padding is used in display objects and it will get confusing if we will use it here as well, especially, as this shape implements display object.